src/common/lib/client/baserealtime.ts

```diff
@@ -10,17 +10,20 @@ import { ChannelOptions } from '../../types/channel';
 import ClientOptions from '../../types/ClientOptions';
 import * as API from '../../../../ably';
 import { ModulesMap } from './modulesmap';
+import RealtimePresence from './realtimepresence';
 
 /**
  `BaseRealtime` is an export of the tree-shakable version of the SDK, and acts as the base class for the `DefaultRealtime` class exported by the non tree-shakable version.
  */
 class BaseRealtime extends BaseClient {
+  readonly _RealtimePresence: typeof RealtimePresence | null;
   _channels: any;
   connection: Connection;
 
   constructor(options: ClientOptions, modules: ModulesMap) {
     super(options, modules);
     Logger.logAction(Logger.LOG_MINOR, 'Realtime()', '');
+    this._RealtimePresence = modules.RealtimePresence ?? null;
```

Comment:
How is this line different from the following?
``` this._RealtimePresence = modules.RealtimePresence;```

Reply:
It would lead to `_RealtimePresence` being set to `undefined` instead of `null`. As written, `_RealtimePresence` doesn’t accept `undefined` as a value. If your question is "why not make `_RealtimePresence` of type `typeof RealtimePresence | undefined`, then it's just a matter of personal preference; I prefer to use `null` to indicate the absence of a value.